### PR TITLE
Update eslint file to include all environments

### PR DESCRIPTION
**Pull request by bidoubiwa**
Update linting rules so that we can have a gather point for all rules.

@mdubus do you want me to use the [linting rules of the landing](https://github.com/meilisearch/landing/blob/main/.eslintrc) ?

@CaroFG do you want me to update the Vue styling  with your rules?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,16 @@ To fix and format your code, you can use the following command:
...
 
+### Different environments
+
+The `eslintrc.js` files contains rules for the following environments:
+- typescript
+- javascript
+- React
+- Vue
+
+Please copy paste the rules that are relevant to you.
```
Comment (CaroFG) [inline README.md:67]: Copy-paste where? Or do you mean delete what does not apply?

Comment (bidoubiwa) [reply README.md:67]: If you create a project for a demo, you can create an `.eslintrc.js` file with the rules corresponding to your environment. 

In this case the Vue rules

**Comment by CaroFG** [comment]
I don't have any different rules to submit

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
+module.exports = {
+  root: true,
+  overrides: [
+    {
+      /*
+       * REACT
+       */
+      files: ['**/*.js'],
+      env: {
+        es2020: true,
```
Comment (mdubus) [inline .eslintrc.js:10]: es2022 ?

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        browser: true,
+        jest: true,
+      },
+      parser: 'babel-eslint',
+      extends: [
+        'eslint:recommended',
+        'plugin:react/recommended',
+        'plugin:prettier/recommended',
+      ],
+      plugins: ['react', 'import'],
```
Comment (mdubus) [inline .eslintrc.js:22]: What does the `import` plugin do ? 
In the list of plugins I usually have `prettier` also, should we add it here?

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+      parserOptions: {
+        ecmaFeatures: {
+          jsx: true,
+        },
+        ecmaVersion: 2020,
```
Comment (mdubus) [inline .eslintrc.js:27]: Should we use `2022` or `latest`?

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        browser: true,
+        es2020: true,
+        'jest/globals': true,
+        node: true,
+      },
+      extends: [
+        'standard',
+        'plugin:prettier/recommended',
+        'eslint:recommended',
+        'plugin:cypress/recommended',
```
Comment (mdubus) [inline .eslintrc.js:65]: Why do we have `cypress` plugins here in the JS config but not above in the React one?

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+      ],
+      plugins: ['jest'],
```
Comment (mdubus) [inline .eslintrc.js:67]: Do we really need it?

File: .eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+      parserOptions: {
+        ecmaVersion: 2020,
+      },
+      rules: {
+        'cypress/no-unnecessary-waiting': 'off',
```
Comment (mdubus) [inline .eslintrc.js:72]: Same as above, maybe it's not the role of the boilerplate to provide tests rules and plugins

File: package.json [hunk 8]
```diff
@@ -21,9 +21,29 @@
...
+    "@typescript-eslint/parser": "^4.32.0",
+    "@vue/eslint-config-typescript": "^7.0.0",
+    "@vue/eslint-plugin": "^4.2.0",
+    "babel-eslint": "^10.1.0",
     "eslint": "^7.29.0",
-    "eslint-config-prettier": "^8.3.0",
-    "eslint-plugin-prettier": "^3.4.0",
+    "eslint-config-prettier": "^8.1.0",
+    "eslint-config-standard": "^16.0.0",
+    "eslint-plugin-cypress": "^2.11.3",
```
Comment (mdubus) [inline package.json:32]: I think we can remove it

File: package.json [hunk 9]
```diff
@@ -21,9 +21,29 @@
...
+    "eslint-plugin-flowtype": "^5.3.1",
+    "eslint-plugin-import": "^2.22.0",
+    "eslint-plugin-jest": "^24.1.5",
```
Comment (mdubus) [inline package.json:35]: Same here, we can remove it

**Pull request closed without merging**